cw_solana/lib/pending_solana_transaction.dart

```diff
@@ -0,0 +1,38 @@
+import 'package:cw_core/pending_transaction.dart';
+
+class PendingSolanaTransaction with PendingTransaction {
+  final double amount;
+  final String signature;
+  final String destinationAddress;
+
+  PendingSolanaTransaction({
+    required this.amount,
+    required this.signature,
+    required this.destinationAddress,
+  });
+
+  @override
+  String get amountFormatted {
+    String stringifiedAmount = amount.toString();
+
+    if (stringifiedAmount.toString().length >= 6) {
+      stringifiedAmount = stringifiedAmount.substring(0, 6);
+    }
+
+    return stringifiedAmount;
+  }
+
+  @override
+  Future<void> commit() async {}
```

Comment:
this can't be empty, this one should be the function that actually triggers sending/broadcasting the transaction.
I believe you are currently sending the transaction right away, that's why Ramo mentioned that even when the transaction got an error, it broadcasted to the blockchain